pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -277,12 +276,30 @@ func (b *volumeBinder) BindPodVolumes(assumedPod *v1.Pod) error {
 		return err
 	}
 
-	return wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
+	if err := wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
 		// Get cached values every time in case the pod gets deleted
 		bindings = b.podBindingCache.GetBindings(assumedPod, assumedPod.Spec.NodeName)
 		claimsToProvision = b.podBindingCache.GetProvisionedPVCs(assumedPod, assumedPod.Spec.NodeName)
 		return b.checkBindings(assumedPod, bindings, claimsToProvision)
-	})
+	}); err != nil {
+		return err
+	}
+
+	node, err := b.ctrl.kubeClient.CoreV1().Nodes().Get(assumedPod.Spec.NodeName, metav1.GetOptions{})
```

Comment:
Can we avoid making api calls? Maybe we need to plumb through the node informer during initialization.

Reply:
Done.